### DISCUSSION TITLE
Welcome to vscode-circuitpython Discussions!

### DISCUSSION THREAD

**Discussion by joedevivo** [joedevivo/vscode-circuitpython#39] [General]
## 👋 Welcome!
  I built this extension to work the way I do, which is to open the circuit python device's volume as a VSCode workspace. I'm not sure if that was documented anywhere, nor do I know if that's how other people work. 

  I'm hoping that this discussion forum can be more of a dialog about how more people want to use the extension.

**Comment by askpatrickw** [comment]
When you work that way, don't you see the device resetting and soft-loading all the time? On MacOS, that definitely happens. I usually work locally and copy files over to my device.

So locally I'll have m4_mqtt_project.py and then copy it to /Volumes/CircuitPython/code.py

I used to use the sync tools when I did more MicroPython stuff.
https://phixed.co/blog/micropython-workflow/

**Comment by Idirianuk** [comment]
Hi, Joe.

Love the extension. (No update to 1.13 as I write, so it's rather broken at the moment). Looking forward to having it working again, as it's SO useful.

I had a go at trying to find out what was broken in 1.11 and managed to get as far as being able to debug, pretty much by trial and error. Not totally new to coding (spent many years writing games commercially on PC, ST, Amiga, Genesis, and Saturn, but haven't done much coding in over 20 years), so it's all a bit confusing atm.

I even managed to get it to build to a VSIX, but am totally unsure as to how I managed this (I know, take notes!). Would you be willing to put up a little documentation describing the build process you employ? No worries, if not. :)

Also, I believe CircuitPython 7 is in alpha right now. Do you have any plans to take this into account?

Dunno if that's all a bit rambling, but it's bedtime for me over here in the UK and I'm tired. :)

Thanks for your rather excellent efforts. :)

Enjoy your coffees. :)

**Comment by dwitt** [comment]
Hi Joe,

Is there any way of activating the extension in a sub folder that contains code.py or one of the other file names you list? I have a project where I have folders with code for different circuit python boards that all talk to each other.

**Reply by Idirianuk** [reply]
I store all my projects on a hard drive, editing the source code there, then use a file sync. program to copy to my circuitpython devices. An easy way to ensure that you have all the files needed for the extension to work is to copy the contents of a fresh circuitpython device to a folder on your PC, edit those files then sync to your device. As long as the devices are on different COM ports for the repl to work, it all seems to work using separate vs.code instances.
